### ISSUE TITLE
macOS DAW UIs become very unresponsive after plugin window is opened

### ISSUE THREAD

**Issue by jakerr** [RustAudio/baseview#126]
Opening a gui plugin that uses baseview on macOS causes the entire DAW (FL Studio, Reaper) to be very slow to respond especially when resizing windows and panels.

After much debugging I finally came along this line in [src/baseview/src/macos/view.rs](
https://github.com/RustAudio/baseview/blob/master/src/macos/view.rs#L262)

```rust
let _: () = msg_send![class!(NSEvent).setMouseCoalescingEnabled: NO];
```

Because [mouseCoalescingEnabled](https://developer.apple.com/documentation/appkit/nsevent/2870068-mousecoalescingenabled?language=objc) is a class property on `NSEvent` changing it affect the entire process i.e. the host DAW. So from that point forward every mouse event is non coalesced and the number of events is just too great for the various DAWs' GUIs to keep up when moving sliders / resizing windows etc.

I tried removing the line which fixes the slowdown, and the plugin GUI interaction even feels smoother to me without.
I'm not sure why that line was there to begin with though, so If it is indeed required, it would be good to at least set the initial value back once the view is cleaned up.

**Comment by jakerr** [comment]
Another bit of information: This issue wasn't noticeable when using the MacBook (2017) track pad. Using a gaming mouse on the same machine, with high tracking speed, makes the problem **very** apparent - resizing the viewport takes multiple seconds.

**Comment by httnn** [comment]
also just ran into this problem in Reaper and can confirm that removing the mentioned call fixes the issue

**Comment by jakerr** [comment]
This is the chat that happened on Discord when I discovered this:

https://discord.com/channels/590254806208217089/948004564005703740/1012281841832710164

I feel like the consensus was basically that that API probably shouldn't have ever been called but nobody was too sure why it was there in the first place.

**Comment by glowcoil** [comment]
Fixed by #195.
